### PR TITLE
Hide the output columns of a transformer

### LINKED ISSUE #1451: Hide(DeSelect) the Output Columns if the Transformer has Output Streams
If a transformer implements HasOutputStreams, we want to keep the normal output columns for compatibility with the old jobs. 
However, for the normal Output Columns should be hidden.  

image

### PR DESCRIPTION
**Pull request by ClaudiaPHI**
Fixes #1451 
Hide the output columns of a transformer

### CODE REVIEW

**Comment by LosD** [comment]
This seems to have broken

File: engine/core/src/main/java/org/datacleaner/job/builder/TransformerComponentBuilder.java [hunk 1]
```diff
@@ -127,7 +128,17 @@ public TransformerComponentBuilder(AnalysisJobBuilder analysisJobBuilder, Transf
...
                     final String id = _id + "-" + _idGenerator.nextId();
-                    _outputColumns.add(new TransformedInputColumn<>(name, id));
+                    final TransformedInputColumn<Object> column = new TransformedInputColumn<>(name, id);
+                    /*
+                     * If the transformer has the annotation
+                     * "HideOutputColumns", the output columns are hidden in
+                     * other components as input columns. If the user wants to
+                     * use them, he can select them in the transformer.
+                     */
+                    if (this.getDescriptor().getAnnotation(HideOutputColumns.class).isHidden()) {
```
Comment (LosD) [inline engine/core/src/main/java/org/datacleaner/job/builder/TransformerComponentBuilder.java]: Break is here: If the annotation is not there, a NPE is guaranteed.

File: api/src/main/java/org/datacleaner/api/HideOutputColumns.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Determines if the output columns {@link Transformer#getOutputColumns()} of
+ * transformer should hidden in other components
```
Comment (arjansh) [inline api/src/main/java/org/datacleaner/api/HideOutputColumns.java]: ... should be hidden ... instead of ... should hidden ... ?

**Comment by arjansh** [comment]
One tiny comment regarding a missing word in the javadoc. Outside of that it looks good to me!

File: api/src/main/java/org/datacleaner/api/HideOutputColumns.java (hunk 2 above)
Comment (LosD) [inline api/src/main/java/org/datacleaner/api/HideOutputColumns.java]: Yeah. Can you make a quick fix, then merge?

**Pull request merged**